common/stats/status_server.cpp

```diff
@@ -122,8 +124,21 @@ StatusServer::StatusServer(uint16_t port, EndPointToOPMap op_map,
   // dump_heap
   op_map_.emplace("/dump_heap",
                   [] (const Arguments*) {
-                    auto ret = mallctl("prof.dump", nullptr, nullptr, nullptr, 0);
-                    return strerror(ret);
+                    static const char* kMallctlProfDumpNamespace = "prof.dump";
+                    static const std::string kHeapTmpFileTemplate = "/tmp/pprof.heap.{}";
+
+                    std::string tmp_file = folly::sformat(kHeapTmpFileTemplate, static_cast<int>(getpid()));
+                    const char* tmp_file_c_str = tmp_file.c_str();
+                    auto ret = mallctl(kMallctlProfDumpNamespace, nullptr, nullptr, &tmp_file_c_str, sizeof(const char*));
```

Comment:
does heap dump kill the process? 
If not, if dump the same process multiple times, won't the dumps collide with the output file name?